pipeline/matching/core/notifier.py

```diff
@@ -0,0 +1,13 @@
+from typing import List
+
+from pipeline.types import NotificationInfo, NotificationInput
+
+
+class Notifier:
```

Comment:
Can you add some documentation for the class and the method?

```
class Notifier:
    """"
    Write a description of the parent class here.
    
    And any notes that other classes that inherit from this class should know about.
    """

    def __init__(self, name: str):
        self.name = name

    def get_notifications(
        self, inp: NotificationInput
    ) -> List[NotificationInfo]:
        """"
        Write a description of this method here.
        """"
        pass
```

Explain that:
- A notifier only works on one chat at a time
- The return type of `get_notifications()` is a list because there could be more than one notification generated per chat, it could also be empty